aiographite/aiographitesend.py

```diff
@@ -0,0 +1,449 @@
+									metric1 : value1,      // type ( string: int )
+									metric2 : value2, 
+									...
+							  }
+
+			metric1 (metric2, ...) are valid metric name for Graphite
+		"""
+		self.send_valid_dataset_list(dataset.items(), timestamp)
+
+
+
+	def generate_message_for_data_list(self, dataset, timestamp, formate_function, generate_message_function):
+		"""
+			generate proper formatted message 
+			@param:
+			Support two kinds of dataset
+				1)	dataset = [(metric1, value1), (metric2, value2), ...] 
+				or 
+				2)	dataset = [(metric1, value1, timestamp1), (metric2, value2, timestamp2), ...]
+		"""
+		listofData = []
+		for data in dataset:
+			# unpack metric data
+			if len(data) == 2:
+				(metric, value) = data
+			else:
+				(metric, value, data_timestamp) = data
+				timestamp = data_timestamp
+			listOfData.append(formate_function(metric, value, timestamp))
+		message =  generate_message_function(listofData)
+		return message
+
+
+
+	def generate_message_for_plaintext(self, listOfPlaintext):
+		"""
+			return the required message formate for protocol 'plaintext'
+			@param: 
+				listOfPlaintext: ["metric1 value1 timestamp1", "metric2 value2 timestamp2", ...]
+				type: List of String
+		"""
+		return "".join(listOfPlaintext)
+
+
+
+	def generate_message_for_pickle(self, listOfMetricTuples):
+		"""
+			@param: 
+				listOfMetricTuples: [(metric1, (timestamp1, value1), (metric2, (timestamp2, value2), ...]
+		"""
+		payload = pickle.dumps(listOfMetricTuples, protocol=2)
+		header = struct.pack("!L", len(payload))
+		message = header + payload
+		return message
+
+
+	def plaintext_protocol_formatted_data(self, metric, value, timestamp):
+		"""
+			@return: required data formate when sending data through 'plaintext' protocol
+			@return_type: String
+		"""
+		formatted_data = " ".join([metric, str(value), str(timestamp)])
+		return formatted_data + "\n"
+
+
+
+	def pickle_protocol_formatted_data(self, metric, value, timestamp):
+		"""
+			@return: required data formate when sending data through 'pickle' protocol
+			@return_type: Tuple
+		"""		
+		return (metric, (timestamp, value))
+
+
+
+	def send_message(self, message):
+		"""
+			@message: data ready to sent to graphite server
+		"""
+		total_sent = 0
+		message_size = len(message)
+		while total_sent < message_size:
+			try:
+				sent = self.socket.send(message[total_sent:].encode('ascii'))
+				if sent == 0:
+					raise RuntimeError("socket connection broken")
+				total_sent = total_sent + sent
+			except socket.gaierror as e:
+				raise AioGraphiteSendException("Fail to send data to %s, Error: %s" % (self.graphite_server_address, e)) 
+			except Exception as e:
+				raise e
+		return total_sent
+
+
+
+	def disconnect(self):
+		"""
+			Close the TCP connection 
+		"""
+		try:
+			self.socket.shutdown(1)
+		except AttributeError:
+			self.socket = None
+		except Exception:
+			self.socket = None
+		finally:
+			self.socket = None
+
+
+	def close_event_loop(self):
```

Comment:
do we need this call? I would hope consumers would know to close the loop directly, or use asyncio.get_event_loop.close().